File: Testes técnicos/Lumni Educação/2 - Parênteses balanceados/parenteses_balanceados.js
function main(parenteses) {
    let parentesesCount = [];
    
    for (let i = 0; i < parenteses.length; i++) {
        // console.log(parenteses, parentesesCount, "1")
        if (parenteses[i] === "(") {
             parentesesCount.push(parenteses[i]);
        }
        
        else if (parenteses[i] === ")") {
            if (parentesesCount[parentesesCount.length - 1] === "("){
                // console.log(parentesesCount[parentesesCount.length - 1], 2);
                parentesesCount.pop();
            }
            
            else {
                parentesesCount.push("#");
            }
        }
    }
   console.log(parentesesCount.length);
   return parentesesCount.length;   
}


let a = main("()))");
let b = main("()()");
let c = main("())(");

File: Testes técnicos/Lumni Educação/2 - Parênteses balanceados/parenteses_balanceados.java
import java.util.*;
import java.lang.*;
import java.io.*;

// The main method must be in a class named "Main".
class Main {
    public static void main(String[] args) {
        String s =  new String("()()");;
        Stack<Character> stack = new Stack<>();
        
        for(int i = 0; i < s.length(); i++) {
            char letra = s.charAt(i);
            
            if(letra == '(') {
                stack.push(letra);
            }
            
            else {
                if(stack.size() > 0 && stack.peek() == '(') {
                    stack.pop();
                }
                
                else {
                    stack.push(letra);
                }
            }
        }
        
        System.out.println(stack.size());
    }
}
